test/e2e/features/composition.feature

```diff
@@ -0,0 +1,193 @@
+Feature: Use Composite Resources to build own APIs
+
+  As a user of Crossplane
+  I want to Use Composite Resources to build my own platform with opinionated concepts and APIs
+  without needing to write a Kubernetes controller from scratch.
+
+  Background:
+    Given Crossplane is running in cluster
+
+  Scenario: Create managed resources through claim creation
+    Given provider xpkg.upbound.io/upbound/provider-dummy:v0.3.0 is running in cluster
+    And CompositeResourceDefinition is present
+      """
+        apiVersion: apiextensions.crossplane.io/v1
+        kind: CompositeResourceDefinition
+        metadata:
+          name: xrobots.dummy.crossplane.io
+          labels:
+            provider: dummy-provider
+        spec:
+          defaultCompositionRef:
+            name: robots-test
+          group: dummy.crossplane.io
+          names:
+            kind: XRobot
+            plural: xrobots
+          claimNames:
+            kind: Robot
+            plural: robots
+          versions:
+            - name: v1alpha1
+              served: true
+              referenceable: true
+              schema:
+                openAPIV3Schema:
+                  type: object
+                  properties:
+                    spec:
+                      type: object
+                      properties:
+                        color:
+                          type: string
+                      required:
+                        - color
```

Comment:
Sure, we could make a step that reads from filesystem, anything is possible because we author what step logic should be. I can tell you that in the projects where I used Cucumber (e.g. [Service Binding Operator](https://github.com/redhat-developer/service-binding-operator)), we never had issues with indentation. Furthermore, the issues could be easily detected either at the test execution or through [linting](https://github.com/vsiakka/gherkin-lint).

The benefit of having yaml content inline is that it is easy to follow the flow, all test details are in the single place.

Reply:
Complex Composition yamls are a frequent source of obvious understandable human indentation errors. I am not convinced it will be somehow different in between Gherkin strings :) But if it fails fast with e.g. validation webhook enabled on the test cluster, then it is probably fine.

Reply:
Definitely agree with @ytsarev's concerns here.

My personal preference is writing Go structs rather than YAML manifests for the type safety. That said, I can live with manifests and see their advantages. Loading them as test fixtures from a regular `.yaml` file seems like it would allow editors, linters, etc to more easily operate on the file as standard YAML, rather than YAML-inside-Gherkin.